ScoutSuite/cli_parser.py

```diff
@@ -1,264 +1,339 @@
+
+        self._init_common_args_parser()
+
+        self._init_rules_generator_parser()
+        self._init_listall_parser()
+
+        self._init_aws_parser()
+        self._init_gcp_parser()
+        self._init_azure_parser()
+
+    def _init_rules_generator_parser(self):
+        parser = self.subparsers.add_parser("ruleset",
+                                            parents=[self.base_args_parser],
+                                            help="Run the ScoutSuite rules generator")
+        parser.add_argument('--ruleset-name',
+                            dest='ruleset_name',
+                            default=None,
+                            required=True,
+                            help='Name of the ruleset to be generated.')
+        parser.add_argument('--base-ruleset',
+                            dest='base_ruleset',
+                            default='default',
+                            help='Ruleset to be used as the baseline.')
+        parser.add_argument('--rules-dir',
+                            dest='rules_dir',
+                            default=[],
+                            nargs='+',
+                            help='Path to directories where custom rules are defined.')
+        parser.add_argument('--generator-dir',
+                            dest='generator_dir',
+                            default=DEFAULT_REPORT_DIR,
+                            help='Path of the Scout2 rules generator.')
+        parser.add_argument('--no-browser',
+                            dest='no_browser',
+                            default=False,
+                            action='store_true',
+                            help='Do not automatically open the report in the browser.')
+
+    def _init_listall_parser(self):
+        parser = self.subparsers.add_parser("listall",
+                                            parents=[self.base_args_parser],
+                                            help="Run the ScoutSuite CSV exporter")
+
+        default = os.environ.get('AWS_PROFILE', 'default')
+        default_origin = " (from AWS_PROFILE)." if 'AWS_PROFILE' in os.environ else "."
+        parser.add_argument('--profile',
+                            dest='profile',
+                            default=[default],
+                            nargs='+',
+                            help='Name of the profile. Defaults to %(default)s' + default_origin)
+        parser.add_argument('--report-dir',
+                            dest='report_dir',
+                            default=DEFAULT_REPORT_DIR,
+                            help='Path of the ScoutSuite report.')
+        parser.add_argument('--ip-ranges',
+                            dest='ip_ranges',
+                            default=[],
+                            nargs='+',
+                            help='Config file(s) that contain your known IP ranges')
+        parser.add_argument('--timestamp',
```

Comment:
This argument is also in the common parser, do we need it here as well?

Reply:
Yep, as Listall doesn't use the common parser. I believed it was cleaner to duplicate this one than to create another parser for a single argument

Reply:
Sounds good!